File: core/src/nl/t64/game/rpg/screens/loot/LootSlotsTableListener.java
package nl.t64.game.rpg.screens.loot;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.InputListener;
import lombok.AllArgsConstructor;
import nl.t64.game.rpg.constants.Constant;

import java.util.function.IntConsumer;


@AllArgsConstructor
class LootSlotsTableListener extends InputListener {

    private final Runnable closeScreenFunction;
    private final Runnable toggleTooltip;
    private final Runnable takeItemFunction;
    private final IntConsumer selectNewSlot;
    private final int slotsInRow;

    @Override
    public boolean keyDown(InputEvent event, int keycode) {
        switch (keycode) {
            case Constant.KEYCODE_RIGHT,
                    Input.Keys.ESCAPE -> closeScreenFunction.run();
            case Constant.KEYCODE_SELECT,
                    Input.Keys.T -> toggleTooltip.run();
            case Constant.KEYCODE_BOTTOM,
                    Input.Keys.ENTER,
                    Input.Keys.A -> takeItemFunction.run();
            case Input.Keys.UP -> selectNewSlot.accept(-slotsInRow);
            case Input.Keys.DOWN -> selectNewSlot.accept(slotsInRow);
            case Input.Keys.LEFT -> selectNewSlot.accept(-1);
            case Input.Keys.RIGHT -> selectNewSlot.accept(1);
        }
        return true;
    }

}

File: core/src/nl/t64/game/rpg/screens/loot/LootSlotsTableListener.kt
package nl.t64.game.rpg.screens.loot

import com.badlogic.gdx.Input
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.InputListener
import nl.t64.game.rpg.constants.Constant


internal class LootSlotsTableListener(
    private val closeScreenFunction: () -> Unit,
    private val toggleTooltip: () -> Unit,
    private val takeItemFunction: () -> Unit,
    private val selectNewSlot: (Int) -> Unit,
    private val slotsInRow: Int
) : InputListener() {

    override fun keyDown(event: InputEvent, keycode: Int): Boolean {
        when (keycode) {
            Constant.KEYCODE_RIGHT, Input.Keys.ESCAPE -> closeScreenFunction.invoke()
            Constant.KEYCODE_SELECT, Input.Keys.T -> toggleTooltip.invoke()
            Constant.KEYCODE_BOTTOM, Input.Keys.ENTER, Input.Keys.A -> takeItemFunction.invoke()
            Input.Keys.UP -> selectNewSlot.invoke(-slotsInRow)
            Input.Keys.DOWN -> selectNewSlot.invoke(slotsInRow)
            Input.Keys.LEFT -> selectNewSlot.invoke(-1)
            Input.Keys.RIGHT -> selectNewSlot.invoke(1)
        }
        return true
    }

}
